### PR TITLE
feat: Add per-class alpha tensor support to FocalLoss

### PR DESCRIPTION
**Pull request by Ahmed-Panaverse**
## Description
Adds support for per-class alpha weighting in FocalLoss while maintaining backward compatibility. It will help to choose dynamic alpha based on class imbalance

## Motivation
Enables fine-grained class balancing for imbalanced datasets while maintaining the existing API.

## Performance
- Tensor alpha adds <1% overhead vs scalar alpha
- Memory impact: O(num_classes) for alpha storage

## Changes
- Modified `alpha` parameter to accept `Union[float, Tensor]`
- Updated type hints and documentation
- Added comprehensive test coverage
- Maintained backward compatibility with scalar alpha

## Testing
- All existing tests pass
- New tests verify tensor alpha functionality

Note: Existing code using `alpha: float` continues to work unchanged.

## Example Usage
```python
# Per-class weighting
alpha = torch.tensor([0.2, 0.3, 0.5])  # For 3 classes
criterion = FocalLoss(alpha=alpha, gamma=2.0)

### CODE REVIEW

File: kornia/losses/focal.py [hunk 1]
```diff
@@ -34,7 +34,8 @@
 def focal_loss(
     pred: Tensor,
     target: Tensor,
-    alpha: Optional[float],
+    alpha: Optional[Union[float, Tensor]],  # Changed to accept Tensor
+    # alpha: Optional[float],
```
Comment (edgarriba) [inline kornia/losses/focal.py]: remove commented code

**Review by edgarriba — CHANGES_REQUESTED** [review]
remove commented codes

File: kornia/losses/focal.py (hunk 1 above)
Comment (Ahmed-Panaverse) [reply kornia/losses/focal.py]: I did :D

Comment (Ahmed-Panaverse) [reply kornia/losses/focal.py]: any other suggestions? Shall I remove all the commented code? I kept it for backward compatibility and to review if required.. I did test this multi loss function for my models locally and it worked perfectly .

**Comment by edgarriba** [comment]
We are doing cleanup. Please, rebase and reopen if you still have interest in merging this PR.
